xdsl/ir.py

```diff
@@ -452,17 +451,22 @@ def create(cls: type[OpT],
                regions: Sequence[Region] | None = None) -> OpT:
         op = Operation.with_result_types(cls, operands, result_types,
                                          attributes, successors, regions)
-        return cast(OpT, op)
+        return op
 
     @classmethod
-    def build(cls: type[OpT],
-              operands: list[Any] | None = None,
-              result_types: list[Any] | None = None,
-              attributes: dict[str, Any] | None = None,
-              successors: list[Any] | None = None,
-              regions: list[Any] | None = None) -> OpT:
+    def build(
+        cls: type[OpT],
+        operands: list[SSAValue | Operation | list[SSAValue | Operation]]
+        | None = None,
+        result_types: list[Any] | None = None,
+        attributes: dict[str, Attribute | Any] | None = None,
```

Comment:
The `Attribute | Any` type hint seems quite weird to me. Attribute is contained in Any, right? Why mention it explicitely?

Reply:
yeah, I agree, Any should be avoided as much as possible

Reply:
This is an opportunity to list out exactly which types the irdl magic happens for, so it tells you early that you gave it something it doesn't support, as opposed to an arcane IRDL error